File: frontend/src/types/assessment.ts
export interface QuestionOption {
  label: string;
  sub_domain?: string;
  score_vector?: Record<string, number>;
}

export interface Question {
  id: string;
  category: 'personality' | 'skill' | 'interest';
  sub_domain?: string;
  question_text: string;
  options: QuestionOption[];
  order_index: number;
}

export interface AssessmentStartResponse {
  assessment_id: string;
  status: string;
  questions: Question[];
  message: string;
}

export interface SingleAnswerSubmission {
  question_id: string;
  selected_option_index: number;
}

export interface AssessmentSubmitRequest {
  assessment_id: string;
  answers: SingleAnswerSubmission[];
}

export interface CategoryScores {
  Analytical: number;
  Technical: number;
  Creative: number;
  Leadership: number;
  Collaborative: number;
  [key: string]: number;
}

export interface AssessmentResult {
  id: string;
  assessment_id: string;
  user_id: string;
  category_scores: CategoryScores;
  top_traits: string[];
  personality_type: string;
  recommended_domains: string[];
  summary: string;
  created_at: string;
}

File: backend/app/schemas/assessment.py
"""
OneStop AI - Assessment Schemas
Pydantic models for assessment questions, answers, sessions, and results validation.
"""

from pydantic import BaseModel, Field
from datetime import datetime


class QuestionOptionSchema(BaseModel):
    label: str
    sub_domain: str | None = None
    score_vector: dict[str, int] = Field(default_factory=dict)


class QuestionResponseSchema(BaseModel):
    id: str
    category: str
    sub_domain: str | None = None
    question_text: str
    options: list[QuestionOptionSchema]
    order_index: int

    class Config:
        from_attributes = True


class AssessmentStartResponse(BaseModel):
    assessment_id: str
    status: str
    questions: list[QuestionResponseSchema]
    message: str


class SingleAnswerSubmission(BaseModel):
    question_id: str
    selected_option_index: int


class AssessmentSubmitRequest(BaseModel):
    assessment_id: str
    answers: list[SingleAnswerSubmission]


class AssessmentResultResponse(BaseModel):
    id: str
    assessment_id: str
    user_id: str
    category_scores: dict[str, int]
    top_traits: list[str]
    personality_type: str | None = None
    recommended_domains: list[str]
    summary: str
    created_at: datetime

    class Config:
        from_attributes = True
